feat: product context with propagateActive as input

cesarGamaSa opened the pull request:
<!--
## PR Checklist
Please check if your PR fulfills the following requirements:

[ ] The commit message follows our guidelines: https://github.com/intershop/intershop-pwa/blob/develop/CONTRIBUTING.md
[ ] Tests for the changes have been added (for bug fixes / features)
[ ] Docs have been added / updated (for bug fixes / features)
[ ] Visual changes have been approved by VD / IAD (if applicable)
-->

## PR Type

<!--
What kind of change does this PR introduce?
Please check the one that applies to this PR using "x".
-->

[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[ ] Build-related changes
[ ] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: <!--Please describe.-->

## What Is the Current Behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Propagation is done when changes are applied which include the propagateActive flag.

## What Is the New Behavior?

propagateActive can also be changed as an input of the product context directive, which affects the propagation timing and availability.

## Does this PR Introduce a Breaking Change?

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

[ ] Yes
[X] No

## Other Information

MaxKless commented:
Hi @cesarGamaSa, thanks for the PR. 
I see that this change is just one line in the product context directive. I assume you are using this somewhere in your project and need this change to make your life easier there?
Is there somewhere you can use this new change in a template in the standard PWA? Because as of now, this would be kind of like merging "dead code" that we don't use ourselves.

I'm not saying I won't merge this if you can't find a location I'm just curious :)

cesarGamaSa commented:
Hello @MaxKless. In our project we added this because we found a huge utility in it. We've made a custom quick order which uses product context and two-way data binding for SKU and quantity. In order to enable the add to cart button only when we have at least an SKU with some quantity, we added this and it worked perfectly and made this implementation a lot easier. So we thought that this could also be useful for the standard PWA, ence my PR.
